dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -0,0 +1,339 @@
+package definitions
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"errors"
+
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/loadbalancer"
+	"gopkg.in/yaml.v2"
+)
+
+type RouteGroupValidator struct {
+	FilterRegistry filters.Registry
+}
+
+var rgv = &RouteGroupValidator{}
+
+func (rgv *RouteGroupValidator) Validate(item *RouteGroupItem) error {
+	err := rgv.BasicValidation(item)
+	if rgv.FilterRegistry != nil {
+		err = errors.Join(err, rgv.FiltersValidation(item))
+	}
+	return err
+}
+
+func (rgv *RouteGroupValidator) BasicValidation(item *RouteGroupItem) error {
+	return item.validate()
+}
+
+func (rgv *RouteGroupValidator) FiltersValidation(item *RouteGroupItem) error {
+	return validateRouteGroupFilters(item, rgv.FilterRegistry)
+}
+
+func (rgv *RouteGroupValidator) ValidateList(rl *RouteGroupList) error {
+	err := rgv.BasicValidationList(rl)
+
+	if rgv.FilterRegistry != nil {
+		err = errors.Join(err, rgv.FiltersValidationList(rl))
+	}
+	return err
+}
+
+func (rgv *RouteGroupValidator) BasicValidationList(rl *RouteGroupList) error {
+	var err error
+	// avoid the user having to repeatedly validate to discover all errors
+	for _, i := range rl.Items {
+		err = errors.Join(err, rgv.BasicValidation(i))
+	}
+	return err
+}
+
+func (rgv *RouteGroupValidator) FiltersValidationList(rl *RouteGroupList) error {
+	var err error
+	// avoid the user having to repeatedly validate to discover all errors
+	for _, i := range rl.Items {
+		err = errors.Join(err, rgv.FiltersValidation(i))
+	}
+	return err
+}
+
+// ParseRouteGroupsJSON parses a json list of RouteGroups into RouteGroupList
+func ParseRouteGroupsJSON(d []byte) (RouteGroupList, error) {
+	var rl RouteGroupList
+	err := json.Unmarshal(d, &rl)
+	return rl, err
+}
+
+// ParseRouteGroupsYAML parses a YAML list of RouteGroups into RouteGroupList
+func ParseRouteGroupsYAML(d []byte) (RouteGroupList, error) {
+	var rl RouteGroupList
+	err := yaml.Unmarshal(d, &rl)
+	return rl, err
+}
+
+// UnmarshalJSON creates a new skipperBackend, safe to be called on nil pointer
+func (sb *SkipperBackend) UnmarshalJSON(value []byte) error {
```

Comment:
I think this json/yaml heloers do not belong to validation, lets move them back routegroups.go